packages/fiori/src/DynamicPageTitle.ts

```diff
@@ -0,0 +1,287 @@
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	snappedHeading!: HTMLElement[];
+
+	/**
+	 * Defines the heading that is shown only when the header is expanded.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	expandedHeading!: HTMLElement[];
+
+	/**
+	 * Defines the actions in the Dynamic page title.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	actions!: HTMLElement[];
+
+	/**
+	 * Defines the navigation actions in the Dynamic page title.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	navigationActions!: HTMLElement[];
+
+	/**
+	 * Defines the content of the Dynamic page title.
+	 *
+	 * @public
+	 */
+	@slot({ "default": true, type: HTMLElement })
+	content!: HTMLElement[];
+
+	/**
+	 * Defines the content of the title that is shown only when the header is snapped.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	snappedContent!: HTMLElement[];
+
+	/**
+	 * Defines the content of the title that is shown only when the header is expanded.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	expandedContent!: HTMLElement[];
+
+	/**
+	 * Defines the content of the breadcrumbs inside Dynamic Page Title.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	breadcrumbs!: HTMLElement[];
+
+	static i18nBundle: I18nBundle;
+
+	_handleResize: ResizeObserverCallback;
+	minContentWidth?: number;
+	minActionsWidth?: number;
+
+	constructor() {
+		super();
+		this._handleResize = this.handleResize.bind(this);
+	}
+
+	static async onDefine() {
+		DynamicPageTitle.i18nBundle = await getI18nBundle("@ui5/webcomponents-fiori");
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._handleResize);
+		if (isDesktop()) {
+			this.setAttribute("desktop", "");
+		}
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._handleResize);
+	}
+
+	onBeforeRendering() {
+		this.prepareLayoutActions();
+	}
+
+	get styles() {
+		return {
+			content: {
+				"min-width": `${this.minContentWidth || 0}px`,
+			},
+			actions: {
+				"min-width": `${this.minActionsWidth || 0}px`,
+			},
+		};
+	}
+
+	get hasContent() {
+		return !!this.content.length;
+	}
+
+	get hasHeading() {
+		return !!this.heading.length;
+	}
+
+	get headingSlotName() {
+		if (this.hasHeading) {
+			return "heading";
+		}
+		if (!this.snapped) {
+			return "expandedHeading";
+		}
+		return "snappedHeading";
+	}
+
+	get contentSlotName() {
+		return !this.snapped ? "expandedContent" : "snappedContent";
+	}
+
+	get _headerExpanded() {
+		return !this.snapped;
+	}
+
+	get _ariaDescribedbyText() {
+		return DynamicPageTitle.i18nBundle.getText(DYNAMIC_PAGE_ARIA_DESCR_TOGGLE_HEADER);
+	}
+
+	get _ariaLabelledBy() {
+		const hasAnyHeading = this[this.headingSlotName].length;
+		if (hasAnyHeading) {
+			return `${this._id}-heading`;
+		}
+	}
+
+	prepareLayoutActions() {
+		// all navigation/layout actions should have the NeverOverflow behavior
+		const navigationActions = this.querySelector<Toolbar>("[ui5-toolbar][slot='navigationActions']");
```

Comment:
If you expect toolbar to be assigned to actions / navigationActions it would be good to rename them to `navigationBar/actionsBar` otherwise it will be confused because I would assume that i have to pass ui5-button instead of ui5-toolbar